### PR TITLE
Improvements to the commit processor

### LINKED ISSUE #3532: cvmfs_receiver: avoid final fetch of manifest
cvmfs_receiver makes a final fetch of the manifest for no purpose other than to log the new catalog hash

https://github.com/cvmfs/cvmfs/blob/2333588f3d2afd54e3b145336740539bdf053f9c/cvmfs/receiver/commit_processor.cc#L250-L278

### LINKED ISSUE #3535: cvmfs_receiver: avoid reinitialization of server_tool
downloadManager intialisation does DNS resolution of the proxy hostnames. This can be slow. 
Server_tool gets reinitialised at
https://github.com/cvmfs/cvmfs/blob/2333588f3d2afd54e3b145336740539bdf053f9c/cvmfs/receiver/commit_processor.cc#L210-L212
which implicitly initialises a new downloadmanager. This can be avoided by making server_tool able to init its signing_tool instance for both signing and verifying.  by modifying `ServerTool::InitSigningSignatureManager` and `ServerTool::InitVerifyingSignatureManager` appropriately

### PR DESCRIPTION
**Pull request by jblomer**
Addresses several performance issues when committing catalogs through the gateway. 

There are more inefficiencies around the server tool and the signing tool. We should address them in several follow-up PRs, I think.

Fixes #3532
Fixes #3535

### CODE REVIEW

**Comment by vvolkl** [comment]
@cernvm-bot unittest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Review by vvolkl — APPROVED** [review]
Looks good, but now needs a rebase. Should I just cherry-pick the changes that avoid the manifest download in a new PR?

**Comment by jblomer** [comment]
I'll make a full rebase just to make sure we don't overlook dependencies between the commits.

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtesting

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

File: cvmfs/receiver/commit_processor.cc [hunk 1]
```diff
@@ -250,33 +248,13 @@ CommitProcessor::Result CommitProcessor::Process(
...
-    UniquePtr<manifest::Manifest> manifest(server_tool->FetchRemoteManifest(
-        params.stratum0, repo_name, manifest_base_hash));
-
-    LogCvmfs(kLogReceiver, kLogSyslog,
-             "CommitProcessor - lease_path: %s, new root hash: %s",
-             lease_path.c_str(),
-             manifest->catalog_hash().ToString(false).c_str());
-  }
+  const UniquePtr<manifest::Manifest> manifest_new(
+    manifest::Manifest::LoadFile(new_manifest_path));
```
Comment (vvolkl) [inline cvmfs/receiver/commit_processor.cc]: At this point the manifest file is unfortunately no longer around, this is a tmpfile essentially. An easy fix is to just get the hash directly from the merge tool. I have a patch I can add directly here if you agree @jblomer

File: cvmfs/receiver/commit_processor.cc [hunk 2]
```diff
@@ -138,19 +138,23 @@ CommitProcessor::Result CommitProcessor::Process(
...
+  {
     LogCvmfs(
         kLogReceiver, kLogSyslogErr,
         "CommitProcessor - error: Could not initialize the signature manager");
     return kError;
   }
 
   shash::Any manifest_base_hash;
-  UniquePtr<manifest::Manifest> manifest(server_tool->FetchRemoteManifest(
+  const UniquePtr<manifest::Manifest> manifest_tgt(
```
Comment (HereThereBeDragons) [inline cvmfs/receiver/commit_processor.cc:152]: what does the `tgt` in `manifest_tgt` stand for?

Comment (vvolkl) [reply cvmfs/receiver/commit_processor.cc:152]: I'm assuming "target"

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Pull request merged**